coderd/externalauth/externalauth.go

```diff
@@ -101,6 +110,19 @@ func (c *Config) RefreshToken(ctx context.Context, db database.Store, externalAu
 		// we aren't trying to surface an error, we're just trying to obtain a valid token.
 		return externalAuthLink, false, nil
 	}
+
+	var extra json.RawMessage
+	if len(c.ExtraTokenKeys) > 0 {
+		extraMap := map[string]interface{}{}
+		for _, key := range c.ExtraTokenKeys {
+			extraMap[key] = token.Extra(key)
+		}
+		extra, err = json.Marshal(extraMap)
+		if err != nil {
+			return externalAuthLink, false, xerrors.Errorf("marshal extra token keys: %w", err)
+		}
+	}
```

Comment:
Can we extract this logic to a function to use in both spots? Instead of the duplicate logic?

Reply:
I'll add it to the `externalauth` package